front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -19,66 +18,83 @@ import { useNavigate } from 'react-router-dom';
 import { setSuccess } from 'reducers/main';
 import { updateScenarioID } from 'reducers/osrdconf';
 import { getInfraID, getProjectID, getStudyID } from 'reducers/osrdconf/selectors';
-import { ScenarioListResult } from 'common/api/osrdEditoastApi';
-import {
-  ELECTRICAL_PROFILE_SET_URI,
-  PROJECTS_URI,
-  SCENARIOS_URI,
-  STUDIES_URI,
-} from '../operationalStudiesConsts';
+import { ScenarioListResult, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { ScenarioRequest } from 'common/api/osrdMiddlewareApi';
 
 const scenarioTypesDefaults = {
   name: '',
   description: '',
   infra: undefined,
 };
 
-type Props = {
+type AddOrEditScenarioModalProps = {
   editionMode?: boolean;
   scenario?: ScenarioListResult;
   getScenarioTimetable?: (v: boolean) => void;
 };
 
+type SelectOptionsType = { key: number | undefined; value: string };
+
 export default function AddOrEditScenarioModal({
   editionMode = false,
   scenario,
   getScenarioTimetable,
-}: Props) {
+}: AddOrEditScenarioModalProps) {
   const { t } = useTranslation('operationalStudies/scenario');
   const { closeModal } = useContext(ModalContext);
+  const [deleteScenarioRTK] =
+    osrdEditoastApi.endpoints.deleteProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId.useMutation(
+      {}
+    );
+  const [patchScenario] =
+    osrdEditoastApi.endpoints.patchProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId.useMutation(
+      {}
+    );
+  const [postScenario] =
+    osrdEditoastApi.endpoints.postProjectsByProjectIdStudiesAndStudyIdScenarios.useMutation({});
+
   const noElectricalProfileSetOption = {
     key: undefined,
     value: t('noElectricalProfileSet'),
   };
   const [currentScenario, setCurrentScenario] = useState<ScenarioListResult>(
     scenario || scenarioTypesDefaults
   );
+  const { data: electricalProfils, isError } = osrdEditoastApi.useGetElectricalProfileSetQuery();
+
   const [displayErrors, setDisplayErrors] = useState(false);
-  const [electricalProfileSetOptions, setElectricalProfileSetOptions] = useState([
-    noElectricalProfileSetOption,
-  ]);
+  const [electricalProfileSetOptions, setElectricalProfileSetOptions] = useState<
```

Comment:
You should rather use a useMemo to directly define this variable:
```
  const electricalProfileSetOptions = useMemo(() => {
    if (isError || electricalProfils.length < 2) {
      return [];
    }

    const sortedElectricalProfils = sortBy(electricalProfils, ['name']);
    return [
      noElectricalProfileSetOption,
      ...sortedElectricalProfils.map((option: ElectricalProfileSetType) => ({
        key: option.id,
        value: option.name,
      })),
    ];
  }, [electricalProfils]);
```
Try to use RTK to remove the maximum of state values :)

Reply:
Done !!